src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileServiceTest.kt

```diff
@@ -48,4 +53,70 @@ internal class PrintResponseFileServiceTest {
         inOrder.verify(printResponseProcessingService).processBatchAndPrintResponses(expectedPrintResponses)
         inOrder.verify(sftpService).removeFileFromOutBoundDirectory(directory, fileName)
     }
+
+    @Test
+    fun `should log and not throw exception if file is not found when deleting`() {
+        // Given
+        val directory = "EROP/Dev/OutBound"
+        val fileName = "status-20221101171156056.json"
+        val fileContent = "{\"BatchResponses\": [], \"PrintResponses\": []}"
+
+        given(sftpService.fetchFileFromOutBoundDirectory(any(), any())).willReturn(fileContent)
+        val expectedPrintResponses = PrintResponses().withBatchResponses(emptyList()).withPrintResponses(emptyList())
+        given(objectMapper.readValue(fileContent, PrintResponses::class.java))
+            .willReturn(expectedPrintResponses)
+        val exception =
+            IOException("Failed to remove file", SftpException(ChannelSftp.SSH_FX_NO_SUCH_FILE, "No such file"))
+        given(sftpService.removeFileFromOutBoundDirectory(any(), any())).willThrow(exception)
+        TestLogAppender.reset()
+
+        // When
+        printResponseFileService.processPrintResponseFile(directory, fileName)
+
+        // Then
+        val inOrder = inOrder(sftpService, objectMapper, printResponseProcessingService)
+        inOrder.verify(sftpService).fetchFileFromOutBoundDirectory(directory, fileName)
+        inOrder.verify(objectMapper).readValue(fileContent, PrintResponses::class.java)
+        inOrder.verify(printResponseProcessingService).processBatchAndPrintResponses(expectedPrintResponses)
+        inOrder.verify(sftpService).removeFileFromOutBoundDirectory(directory, fileName)
+        assertThat(
```

Comment:
I think we've been round this before haven't we? Rather than writing tests that depend on log messages (feels very fragile?) wouldn't it be better to assert that `printDetailsRepository.updateItems` was not called?

Reply:
@NathanRussellValtech , I believe this test is testing the failure in the final step when the responses are processed (sent to queue for processing and eventually will be processed), but the response file cannot be deleted, because it is not found.
 
In this case, the data should be updated with the appropriate statuses and since we don't throw any exceptions the only difference from the happy path will be the warning log. We assert there are no exceptions thrown and I also added the assertion for the warning log.

Reply:
Ah, yes, I see
It's fine as you have it, but I would argue that in this case do we really care? If we didn't log the warning in the impl would it affect the behaviour of the class?
Let's leave it as you have it, because we need to get things merged ASAP so we can move on 👍 (but personally I don't like these tests that assert log messages ¯\_(ツ)_/¯ )